### PR TITLE
CHANGE TO DECORATOR PATTERN

### PR DESCRIPTION
**Pull request by carreraprogrammer**
## PROJECT REQUIREMENTS

- [x]  Check the usage of [Decorator pattern in Ruby](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0)
- [x]  Think about how you can use two decorators in order to capitalize and trim people's names.
In the following steps you will get instructions for implementing the Decorator design pattern for this case.

### Interface 

- Create a class Nameable.
- [x]  Implement a method called correct_name that will raise a NotImplementedError.

### Turn your Person class to Nameable

- [x] Make sure that your Person class inherits from Nameable
- [x] Make sure that this class has a method correct_name implemented. It should simply return the name attribute.

### Prepare base Decorator

- [x] Make sure that it inherits from Nameable.
- [x] In the constructor assign a nameable object from params to an instance variable.
- [x] Implement the correct_name method that returns the result of the correct_name method of the @nameable.

### Prepare CapitalizeDecorator and TrimmerDecorator

- For the CapitalizeDecorator:
- [x] Create a class that inherits from the base Decorator class.
- [x] Implement a method correct_name that capitalizes the output of @nameable.correct_name.

- For the TrimmerDecorator:

- [x] Create a class that inherits from the base Decorator class.
- [x] Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### See your decorators in action
Captura desde 2023-06-21 11-45-01

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
```
Comment (zieeco) [inline decorator.rb]: - [x]  Hey, you should add a setter method to your nameable attribute.  This way,  the attribute will be accessible and modifiable  outside the Decorator class
e.g.

 `attr_accessor :nameable`

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @carreraprogrammer,





Great job so far!  👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice class decoration ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions
- N/A


**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decorator.rb (hunk 1 above)
Comment (carreraprogrammer) [reply decorator.rb]: Hello @zieeco, thanks for the review, changes are done !! 😁

**Review by alexander16108 — APPROVED** [review]
Hi @carreraprogrammer , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Correct use of Flow
✅ Descriptive Pull Request

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by carreraprogrammer** [comment]
Thanks @alexander16108, I will merge it 🎊😁

giphy

**Pull request merged**